com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java

```diff
@@ -562,7 +564,6 @@ private synchronized void stopRetryTimer() {
             timerTask.cancel();
             timerTask = null;
         }
-        retries = 0;
```

Comment:
Because this is a bug. You are starting a new retrytimer in a thread and are setting the **global** retries to 0 here.

Therefore the `retries` is always 1 here: https://github.com/zsmartsystems/com.zsmartsystems.zigbee/blob/master/com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/ash/AshFrameHandler.java#L583

And thus the `disconnect` method will never be called and the queue is just filling up.

Reply:
I am 100% sure that it was always 1 and the `disconnect` was never ever called. So there is a bug in there.

Also I am resetting the counter to 0 in `disconnect`.

Reply:
FYI: Follow-up issue created: https://github.com/zsmartsystems/com.zsmartsystems.zigbee/issues/466